packages/flutter_tools/lib/src/version.dart

```diff
@@ -130,17 +130,34 @@ class FlutterVersion {
   };
 
   /// A date String describing the last framework commit.
-  String get frameworkCommitDate => _latestGitCommitDate();
+  String get frameworkCommitDate => _latestGitCommitDate(lenient: true);
 
-  static String _latestGitCommitDate([ String branch ]) {
+  static String _latestGitCommitDate({
+    String branch,
+    bool lenient = false,
+  }) {
     final List<String> args = gitLog(<String>[
       if (branch != null) branch,
       '-n',
       '1',
       '--pretty=format:%ad',
       '--date=iso',
     ]);
-    return _runSync(args, lenient: false);
+    try {
+      // Don't plumb 'lenient' through directly so that we can print an error
+      // if something goes wrong.
+      return _runSync(args, lenient: false);
+    } on VersionCheckError catch (e) {
+      if (lenient) {
+        final DateTime dummyDate = DateTime.fromMillisecondsSinceEpoch(0);
```

Comment:
Should this be in the future so we don't prompt the new version available warning?

Reply:
Nope I have this backwards, having it in the past will prevent that warning. Ignore!

Reply:
`_latestGitCommitDate()` is used to get both the local and remote date of the latest commit. Luckily, the logic to decide whether to show the update message is already handling `VersionCheckError`. When coming from that logic, `lenient` is false, and the exception will still be thrown. I'll add some comments about this.

Reply:
Investigated this a bit more and added some code to more defensively bail from the update check on a version check failure.